### PR TITLE
add unsubscribe route

### PR DESCRIPTION
**Pull request by ErinLMoore**
Asana ticket: https://app.asana.com/0/1167196461144361/1206700151817926/f

### CODE REVIEW

File: apps/concierge_site/lib/controllers/unsubscribe_controller.ex [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    # encrypting passwords and don't need the extra security
+    {:ok, user_id} =
+      Plug.Crypto.decrypt(conn.secret_key_base, conn.secret_key_base, encrypted_user_id,
+        max_age: @thirty_days_in_seconds
+      )
+
+    Trip.get_trips_by_user(user_id)
+    |> Enum.each(&Trip.pause(&1, user_id))
+
+    json(conn, %{status: "ok"})
```
Comment (arkadyan) [inline apps/concierge_site/lib/controllers/unsubscribe_controller.ex:17]: thought: Interesting that nothing seems to describe the structure of the expected response. (This seems as fine as anything.)

File: apps/concierge_site/test/web/controllers/unsubcribe_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+defmodule ConciergeSite.UnsubscribeControllerTest do
+  @moduledoc false
+  use ConciergeSite.ConnCase, async: true
+  import AlertProcessor.Factory
+  alias AlertProcessor.Model.Subscription
+  alias AlertProcessor.Repo
+
+  describe "update/2" do
+    test "unsubscribes user from al subscriptions", %{conn: conn} do
```
Comment (arkadyan) [inline apps/concierge_site/test/web/controllers/unsubcribe_controller_test.exs:9]:
```suggestion
    test "unsubscribes user from all subscriptions", %{conn: conn} do
```

nitpick: Typo

File: apps/concierge_site/lib/controllers/unsubscribe_controller.ex [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  def update(conn, %{"id" => encrypted_user_id}) do
+    # We're not using a separate salt here because we're not
+    # encrypting passwords and don't need the extra security
+    {:ok, user_id} =
+      Plug.Crypto.decrypt(conn.secret_key_base, conn.secret_key_base, encrypted_user_id,
+        max_age: @thirty_days_in_seconds
+      )
+
+    Trip.get_trips_by_user(user_id)
+    |> Enum.each(&Trip.pause(&1, user_id))
```
Comment (arkadyan) [inline apps/concierge_site/lib/controllers/unsubscribe_controller.ex:15]: suggestion: As a "belt and suspenders" approach, I wonder if it makes sense to also set the user's `communication_mode` to `none`.

File: apps/concierge_site/lib/controllers/unsubscribe_controller.ex [hunk 4]
```diff
@@ -0,0 +1,19 @@
+defmodule ConciergeSite.UnsubscribeController do
+  use ConciergeSite.Web, :controller
+  alias AlertProcessor.Model.Trip
+  @thirty_days_in_seconds 2_592_000
+
+  def update(conn, %{"id" => encrypted_user_id}) do
+    # We're not using a separate salt here because we're not
+    # encrypting passwords and don't need the extra security
+    {:ok, user_id} =
+      Plug.Crypto.decrypt(conn.secret_key_base, conn.secret_key_base, encrypted_user_id,
```
Comment (arkadyan) [inline apps/concierge_site/lib/controllers/unsubscribe_controller.ex:10]: question: In non-test environments, is this `secret_key_base` value being added to the `conn` automatically? It looks like it's also an application environment variable, but if it's part of the conn, too, that's fine, I'm just not familiar with it.

**Review by arkadyan — APPROVED** [review]
A small suggestion and question, but otherwise looks good.

File: apps/concierge_site/lib/controllers/unsubscribe_controller.ex (hunk 4 above)
Comment (ErinLMoore) [reply apps/concierge_site/lib/controllers/unsubscribe_controller.ex:10]: It apparently does read the secret_key_base out of the config (if available) and puts it in the conn automatically, so I could get it from either. I can't find anything about that in the documentation, but it works fine in my local dev environment.

File: apps/concierge_site/lib/controllers/unsubscribe_controller.ex (hunk 3 above)
Comment (ErinLMoore) [reply apps/concierge_site/lib/controllers/unsubscribe_controller.ex:15]: That's a good idea, I'll put that in.

**Pull request merged**